device.go

```diff
@@ -106,7 +114,7 @@ func newVFIODevice(devInfo DeviceInfo) *VFIODevice {
 	}
 }
 
-func (device *VFIODevice) attach(h hypervisor) error {
+func (device *VFIODevice) attach(h hypervisor, c *Container, create bool) error {
```

Comment:
Forgot to say, can you write tests for `attach()` and `detach()`?

Reply:
Yes, I have updated the tests for block device attach,